clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy

```diff
@@ -16,203 +16,391 @@
-      virtualIP.id = existingPool.vipId
-      openstackClientProvider.updateVip(region, virtualIP)
-      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updated vip ${virtualIP.name} in ${region}."
+  /**
+   * Updates security groups to vip port associated to load balancer.
+   * @param region
+   * @param portId
+   * @param securityGroups
+   */
+  protected void updateSecurityGroups(String region, String portId, List<String> securityGroups) {
+    Port port = clientProvider.getPort(region, portId)
+    if (securityGroups && port.securityGroups != securityGroups) {
+      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updating port ${portId} with security groups ${securityGroups} in ${region}..."
+      clientProvider.updatePort(region, portId, securityGroups)
+      task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updated port ${portId} with security groups ${securityGroups} in ${region}."
     }
+  }
 
-    // Currently only supporting one health check ... Could be extended to support multiple in future
-    String healthMonitorId = existingPool.healthMonitors?.isEmpty() ? null : existingPool.healthMonitors?.first()
-    if (poolHealthMonitor) {
-      if (healthMonitorId) {
-        HealthMonitor healthMonitor = openstackClientProvider.getHealthMonitor(region, healthMonitorId)
-        if (healthMonitor.type.name().equals(poolHealthMonitor.type.name())) {
-          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updating health check ${healthMonitor.id} in ${region}..."
-          // Set id and update
-          poolHealthMonitor.id = healthMonitor.id
-          openstackClientProvider.updateHealthMonitor(region, poolHealthMonitor)
-          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Updated health check ${healthMonitor.id} in ${region}."
-        } else {
-          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Removing existing monitor ${healthMonitorId} and creating health check for lbPool ${existingPool.name} in ${region}..."
-          // Types are different (i.e. PING vs HTTP) and can't be updated
-          openstackClientProvider.disassociateAndRemoveHealthMonitor(region, existingPool.id, healthMonitorId)
-          openstackClientProvider.createHealthCheckForPool(region, existingPool.id, poolHealthMonitor)
-          task.updateStatus UPSERT_LOADBALANCER_PHASE, "Removed existing monitor ${healthMonitorId} and created health check for lbPool ${existingPool.name} in ${region}."
-        }
+  /**
+   * Adds/Removes floating ip address associated to load balancer.
+   * @param region
+   * @param networkId
+   * @param portId
+   */
+  protected void updateFloatingIp(String region, String networkId, String portId) {
+    NetFloatingIP existingFloatingIp = clientProvider.getFloatingIpForPort(region, portId)
+    if (networkId) {
+      Network network = clientProvider.getNetwork(region, networkId)
+      FloatingIP ip = clientProvider.getOrCreateFloatingIp(region, network.name)
```

Comment:
Looks identical to the current solution -> https://github.com/spinnaker/clouddriver/blob/master/clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/ops/loadbalancer/UpsertOpenstackLoadBalancerAtomicOperation.groovy#L181-L210